samples/42.scaleout/src/main/java/com/microsoft/bot/sample/scaleout/MemoryStore.java

```diff
@@ -59,11 +60,12 @@
             }
 
             store.put(key, new Pair<>(content, UUID.randomUUID().toString()));
+            return CompletableFuture.completedFuture(true);
         } catch (InterruptedException e) {
             e.printStackTrace();
-        } finally {
-            semaphore.release();
             return CompletableFuture.completedFuture(true);
```

Comment:
```suggestion
            return CompletableFuture.completedFuture(null);
```

This is just a recommendation as this is a `boolean` return which means that if we could or not save the object.. if we return `true` during the exception it will be weird to the user

Reply:
So, instead of return null, as you suggest, we can return false, right?